api-gateway/security/modules/security-module/src/main/java/io/knotx/examples/security/auth/OAuth2HelloHandlerFactory.java

```diff
@@ -0,0 +1,48 @@
+package io.knotx.examples.security.auth;
+
+import io.knotx.server.api.security.AuthHandlerFactory;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.oauth2.OAuth2ClientOptions;
+import io.vertx.ext.auth.oauth2.OAuth2FlowType;
+import io.vertx.reactivex.core.Vertx;
+import io.vertx.reactivex.ext.auth.oauth2.OAuth2Auth;
+import io.vertx.reactivex.ext.web.Router;
+import io.vertx.reactivex.ext.web.handler.AuthHandler;
+import io.vertx.reactivex.ext.web.handler.OAuth2AuthHandler;
+
+public class OAuth2HelloHandlerFactory implements AuthHandlerFactory {
+
+  private OAuth2AuthHandler oAuth2AuthHandler;

```

Comment:
What if configuration contains two entries for this OAuthHelloHandlerFactory?

Reply:
in `io.knotx.server.SecurityProvider` reference to factories is a map<factoryName, factory reference>, so it will be unique anyway